### ISSUE TITLE
Failing to Compile "The system cannot find the path specified."

### ISSUE THREAD

**Issue by Dracrius** [nnguy132/Switch-Frame-Advancer#1]
I followed the guides linked in the description and have confirmed I have GnuWin32, AVR 8bit Toolchain and LUFA all installed and referenced properly yet I get 3 `The system cannot find the path specified.` messages then it begins to compile and fails. I have one by one moved/deleted the windows path references to verify it is locating all 3 and it will fail with much more obvious error messages as soon as I did so all 3 are being found by the installer so I am quite confused and can not find the place its looking for more paths.

Here is the entire error message 
```
D:\Switch-Frame-Advancer\Switch-Frame-Advancer>make
The system cannot find the path specified.
The system cannot find the path specified.
The system cannot find the path specified.
' [INFO]    :' Begin compilation of project \"Joystick\"...
""
avr-gcc (AVR_8_bit_GNU_Toolchain_3.6.2_1759) 5.4.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

' [GCC]     :' Compiling C file \"Joystick.c\"
avr-gcc -c -pipe -gdwarf-2 -g2 -mmcu=atmega32u4 -fshort-enums -fno-inline-small-functions -fpack-struct -Wall -fno-strict-aliasing -funsigned-char -funsigned-bitfields -ffunction-sections -I. -DARCH=ARCH_AVR8 -DF_CPU=16000000UL -mrelax -fno-jump-tables -x c -Os -std=gnu99 -Wstrict-prototypes -DUSE_LUFA_CONFIG_HEADER -IConfig/  -I. -I./LUFA/LUFA/.. -DARCH=ARCH_AVR8 -DBOARD=BOARD_NONE -DF_USB=16000000UL  -MMD -MP -MF obj/Joystick.d Joystick.c -o obj/Joystick.o
Assembler messages:
Fatal error: can't create obj/Joystick.o: No such file or directory
Joystick.c:296:1: fatal error: error writing to -: Invalid argument
 }
 ^
compilation terminated.
make: *** [obj/Joystick.o] Error 1

```

**Comment by nnguy132** [comment]
I would check your code to see if there is an extra { or } some where near line 296, otherwise try redownloading the code in another file and running the make function on that.

**Comment by nnguy132** [comment]
Do you have avrdude installed?

**Comment by Dracrius** [comment]
> Do you have avrdude installed?

No I installed the AVR 8bit ToolChain from Atmel as suggested, following the instructions on the Splatoon Printer you link. Also i haven't modified your code other then days from 30 to 12 as suggested for my day format. I have also tried compiling the source repo Switch Fightstick and it gives me the same error.

**Comment by Dracrius** [comment]
I just downloaded it to give it a try but it has the least amount of info out of all the tools. Where should I save it for it to be found? Or is it another thing to add to Windows Path?

**Comment by nnguy132** [comment]
http://fab.cba.mit.edu/classes/863.16/doc/projects/ftsmin/windows_avr.html

Following this guide minus the install drivers should work

**Comment by Dracrius** [comment]
> http://fab.cba.mit.edu/classes/863.16/doc/projects/ftsmin/windows_avr.html
> 
> Following this guide minus the install drivers should work

Ah perfect reading that seems adding avrdude to my path is literally the only step I'm missing now. I'll reread that and give it a try when I get back from work now! Thank you

**Comment by Dracrius** [comment]
@nnguy132 unfortunately following that guide and installing avrdude did not help.  it still gives the exact same error. That guide gave me better ways to verify everything was installed and working though with the command line version checks and now I confirmed all are installed correctly as well as I now have avrdude.

**Comment by Dracrius** [comment]
Alright I even went back and looked through everything for more options and noticed the line 'For Linux, follow their instructions for installing the GCC Compiler and Tools. For Windows, you will need the latest AVR toolchain from the Atmel site." while later it talks about the GCC tools so I followed that link which had a link for WinAVR for windows. This is what AVRDude is supposed to replace but I installed it to try that instead. It still fails but at least now it is a different message yet again so the makefile is not using AVRDude correctly it would appear and as the AVRDude site says WinAVR is very  dead. 

Here is what I get with WinAVR installed
```
D:\Switch-Frame-Advancer>make
      0 [main] sh 12800 sync_with_child: child 16256(0x1D8) died before initialization with status code 0xC0000142
    181 [main] sh 12800 sync_with_child: *** child state waiting for longjmp
/usr/bin/sh: fork: Resource temporarily unavailable
      0 [main] sh 14664 sync_with_child: child 13552(0x1E4) died before initialization with status code 0xC0000142
    161 [main] sh 14664 sync_with_child: *** child state waiting for longjmp
/usr/bin/sh: fork: Resource temporarily unavailable
      0 [main] sh 252 sync_with_child: child 6524(0x1E4) died before initialization with status code 0xC0000142
    168 [main] sh 252 sync_with_child: *** child state waiting for longjmp
/usr/bin/sh: fork: Resource temporarily unavailable
 [INFO]    : Begin compilation of project "Joystick"...

avr-gcc (WinAVR 20100110) 4.3.3
Copyright (C) 2008 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

 [GCC]     : Compiling C file "Joystick.c"
avr-gcc -c -pipe -gdwarf-2 -g2 -mmcu=atmega32u4 -fshort-enums -fno-inline-small-functions -fpack-struct -Wall -fno-strict-aliasing -funsigned-char -funsigned-bitfields -ffunction-sections -I. -DARCH=ARCH_AVR8 -DF_CPU=16000000UL -mrelax -fno-jump-tables -x c -Os -std=gnu99 -Wstrict-prototypes -DUSE_LUFA_CONFIG_HEADER -IConfig/  -I. -ILUFA/LUFA/.. -DARCH=ARCH_AVR8 -DBOARD=BOARD_NONE -DF_USB=16000000UL  -MMD -MP -MF obj/Joystick.d Joystick.c -o obj/Joystick.o
Assembler messages:
Fatal error: can't create obj/Joystick.o: No such file or directory
```

Unlike before the console hangs at this point and I can nolonger type new commands.

**Comment by nnguy132** [comment]
Is it possible to have a look at your makefile? Are you using an arduino or teensy with it?

**Comment by nnguy132** [comment]
What program are you using to run the makefile? And what OS are you using?

**Comment by nnguy132** [comment]
Ah youre using WINavr, try using gitbash to run the makefile instead

**Comment by Dracrius** [comment]
GnuWin32 and the command line is how I'm doing it just like all the guides. I only installed WinAVR to test further winAVR is not what runs the make file. I will try and figure out how to do it with bash!

**Comment by Dracrius** [comment]
@nnguy132 The make command doesnt work in bash it just say command not found

**Comment by nnguy132** [comment]
Awesome! Sorry about the slow replies. Im just baffled as to why this is happening

**Comment by Dracrius** [comment]
@nnguy132 I finally got it with getbash. Only reason make stopped working with bash was because I uninstalled WinAVR that software is terrible it cleared my windows path just now. Idk what else was in there but now I look forward to a ton of things breaking as they can't find their paths damn. 

But with GnuWin32, AVR 8bit Toolchain, LUFA and finally AVRDude the only way it will make is through GitBash through command line and Powershell it fails just as described so the current guides are kind of useless for windows. I'd suggest just telling people to install Get for Windows (https://gitforwindows.org/), GnuWin32, AVR 8bit Toolchain, LUFA and finally AVRDude. the guides on the splatoon printer are a mess and far more complicated then need be.
